File: java/Cocinero.java
import java.io.FileWriter;
import java.io.IOException;
import java.util.List;

public class Cocinero extends Thread {
    private final List<Pedido> pedidos;
    private final Object lock;
    private final String logFile;
    private final String nombre;

    public Cocinero(String nombre, List<Pedido> pedidos, Object lock, String logFile) {
        this.nombre = nombre;
        this.pedidos = pedidos;
        this.lock = lock;
        this.logFile = logFile;
    }

    @Override
    public void run() {
        while (true) {
            Pedido pedido = null;

            synchronized (lock) {
                if (!pedidos.isEmpty()) {
                    pedido = pedidos.remove(0);
                } else {
                    break;
                }
            }

            if (pedido != null) {
                prepararPedido(pedido);
            }

            synchronized (lock) {
                registrarLog(pedido);
            }   
        }
    }

    private void prepararPedido(Pedido pedido) {
        System.out.println("[" + nombre + "] recibe " + pedido);
        System.out.println("[" + nombre + "] está preparando " + pedido);
        try {
            Thread.sleep(1000 + (long) (Math.random() * 2000)); // Simula tiempo de preparación
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        System.out.println("[" + nombre + "] ha terminado " + pedido);
    }

    private void registrarLog(Pedido pedido) {
        try (FileWriter fw = new FileWriter(logFile, true)) {
            fw.write("[" + nombre + "] ha preparado " + pedido + "\n");
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: python/cocinero.py
import threading
import time
import random

class Cocinero(threading.Thread):
    def __init__(self, nombre, pedidos, lock, logFile):
        super().__init__()
        self.nombre = nombre
        self.pedidos = pedidos
        self.lock = lock
        self.logFile = logFile

    def run(self):
        while True:
            with self.lock:
                if not self.pedidos:
                    break
                pedido = self.pedidos.pop(0)
            
            self.preparar_pedido(pedido)

            with self.lock:
                self.registrar_log(pedido)

    def preparar_pedido(self, pedido):
        print(f"[{self.nombre}] recibe {pedido}")
        print(f"[{self.nombre}] está preparando {pedido}")
        time.sleep(random.uniform(1, 3))
        print(f"[{self.nombre}] ha terminado {pedido}")
    
    def registrar_log(self, pedido):
        with open(self.logFile, "a", encoding="utf-8") as f:
            f.write(f"[{self.nombre}] ha preparado {pedido}\n")
